doc: provide board porting guidelines

erwango opened the pull request:
Provide board porting guidelines to help achieving a coherent
board library to ease work for people developping application
not board specific and define clear objective to help board
porting reviews.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+Default board configuration
+***************************
+
+When porting a board, you must provide its default configuration. This default
```
Should this be "when porting to a board"?

dbkinder replied:
Or

    When porting to a board, you must provide the board's default
    configuration, used (unless overridden) when an application, test,
    or sample is compiled for this board.

mbolivar replied:
This is verbose. Why not something like:

```
When porting Zephyr to a board, you must provide the board's default
Kconfig configuration, which is used in application builds unless explicitly
overridden.
```

erwango replied:
I'll take this last one, omitting "Kconfig" as it is true for any kind of configuration (pinmux, dts, ..)

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+***************************
+
+When porting a board, you must provide its default configuration. This default
+configuration could be overridden, or otherwise will be the reference when an
```
"the reference one"

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+configuration could be overridden, or otherwise will be the reference when an
+application, test or sample is compiled for this board.
+
+In order to provide coherency accross the various boards and ease the work of
```
consistency better than coherency?

dbkinder replied:
also fix misspelling of ``across``

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+application, test or sample is compiled for this board.
+
+In order to provide coherency accross the various boards and ease the work of
+users providing applications that are not board specific, following guidelines
```
the following guidelines

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+users providing applications that are not board specific, following guidelines
+should be followed when porting a board:
+
+- Provide pins and drivers configuration that matches the board valuable
```
Provide pin and driver configuration that matches...

mbolivar replied:
also "the board's", not "the board"

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+- Provide pins and drivers configuration that matches the board valuable
+components, such as sensors, button, leds or communication i/f such as USB,
+Ethernet connector or Bluetooth/WiFi chip
+- When a well-known connector (arduino, 96board, ....) is present,
```
s/..../etc

dbkinder replied:
- When a well-known connector is present (such as used on an Arduino or 96board),
      configure pins to fit this connector.

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+Ethernet connector or Bluetooth/WiFi chip
+- When a well-known connector (arduino, 96board, ....) is present,
+configure pins to fit this connector
+- Then configure coponents that enable the use of these pins (For instance
```
s/Then//

dbkinder replied:
- Configure components that enable the use of these pins, such as
      configuring and SPI instance for Arduino SPI.

mbolivar replied:
note the "components" typo correction pointed out by @dbkinder

carlescufi commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+configure pins to fit this connector
+- Then configure coponents that enable the use of these pins (For instance
+configure SPI instance for arduino SPI)
+- Configure an output for console
```
the console

dbkinder commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+
+In order to provide coherency accross the various boards and ease the work of
+users providing applications that are not board specific, following guidelines
+should be followed when porting a board:
```
``porting to a board``

dbkinder commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,25 @@ Board
+should be followed when porting a board:
+
+- Provide pins and drivers configuration that matches the board valuable
+components, such as sensors, button, leds or communication i/f such as USB,
```
Make sure the second line of a list item is indented properly
    
    - Provide pin and driver configurations that match the board's
      valuable components such as sensors, buttons, or LEDs, and communication
      interfaces such as USB, Ethernet connector, or Bluetooth/WiFi chip.

erwango commented:
Thanks all for the reviews. Text updated

erwango committed:
doc: provide board porting guidelines

Provide board porting guidelines to help achieving a coherent
board library to ease work for people developing application
not board specific and define clear objective to help board
porting reviews.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mbolivar commented on doc/porting/board_porting.rst:
```diff
@@ -120,3 +120,30 @@ Board
+
+In order to provide consistency across the various boards and ease the work of
+users providing applications that are not board specific, the following
+guidelines should be followed when porting a board:
```
Looks great! The only thing I would add to the end of this sentence is "when possible". Sometimes it is not possible to follow the guidelines, such as if the board does not have a network interface.

mbolivar reviewed (approved):
One minor idea, but I think this looks good.

dbkinder reviewed (approved):
thanks!
